match_realtime.py

```diff
@@ -0,0 +1,233 @@
+
+Configuration settings are stored in configs folder, with compute heavy performance or light-weight alternatives
+available.
+
+Features are saved into a nominated output directory, with one file per image per patch size.
+
+Code is dynamic and can be configured with essentially *any* number of patch sizes, by editing the config files.
+'''
+
+
+import argparse
+import configparser
+import os
+from os.path import join, exists, isfile
+
+from tqdm.auto import tqdm
+import torch
+import torch.nn as nn
+import torchvision.transforms as transforms
+import numpy as np
+import cv2
+from PIL import Image
+from matplotlib import pyplot as plt
+
+
+from patchnetvlad.models.models_generic import get_backend, get_model, get_pca_encoding
+from patchnetvlad.tools.patch_matcher import PatchMatcher
+from patchnetvlad.models.local_matcher import calc_keypoint_centers_from_patches as calc_keypoint_centers_from_patches
+from patchnetvlad.tools import PATCHNETVLAD_ROOT_DIR
+
+
+def input_transform(resize=(480, 640)):
+    if resize[0] > 0 and resize[1] > 0:
+        return transforms.Compose([
+            transforms.Resize(resize),
+            transforms.ToTensor(),
+            transforms.Normalize(mean=[0.485, 0.456, 0.406],
+                                 std=[0.229, 0.224, 0.225]),
+        ])
+    else:
+        return transforms.Compose([
+            transforms.ToTensor(),
+            transforms.Normalize(mean=[0.485, 0.456, 0.406],
+                                 std=[0.229, 0.224, 0.225]),
+        ])
+
+
+def normalise_func(input_diff, num_patches, patch_weights):
+    normed_diff = 0
+    if len(patch_weights) != num_patches:
+        raise ValueError('The number of patch weights must equal the number of patches used')
+    for i in range(num_patches):
+        normed_diff = normed_diff + (patch_weights[i] * input_diff[i])
+    return normed_diff
+
+
+def plot_two(cv_im_one, cv_im_two, inlier_keypoints_one, inlier_keypoints_two):
+
+    kp_all1 = []
+    kp_all2 = []
+    matches_all = []
+    for this_inlier_keypoints_one, this_inlier_keypoints_two in zip(inlier_keypoints_one, inlier_keypoints_two):
+        for i in range(this_inlier_keypoints_one.shape[0]):
+            kp_all1.append(cv2.KeyPoint(this_inlier_keypoints_one[i, 0].astype(float), this_inlier_keypoints_one[i, 1].astype(float), 1, -1, 0, 0, -1))
+            kp_all2.append(cv2.KeyPoint(this_inlier_keypoints_two[i, 0].astype(float), this_inlier_keypoints_two[i, 1].astype(float), 1, -1, 0, 0, -1))
+            matches_all.append(cv2.DMatch(i, i, 0))
+
+    im_allpatch_matches = cv2.drawMatches(cv_im_one, kp_all1, cv_im_two, kp_all2,
+                                          matches_all, None, matchColor=(0, 255, 0), flags=2)
+
+    im_allpatch_matches = cv2.cvtColor(im_allpatch_matches, cv2.COLOR_RGB2BGR)
+
+    cv2.imshow('frame', im_allpatch_matches)
+
+
+def match_two(model, device, opt, config, im_one, im_two):
```

Comment:
And again, import from the existing code, and make more general if needed.

Reply:
This should be possible to do as well, I'll make a new commit with these changes to the pull request